interview-questions.md

```diff
@@ -4,47 +4,47 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. What is object destructuring?
 
-  Your answer:
+  Your answer: It is a method that takes in any number of state objects and allows them to be reffered to by shorthand for the duration of the current method you're in.
 
-  Researched answer:
+  Researched answer: After doing some research I discovered that destructing is a JavaScript feature that allows you to extract multiple pieces of data from an array or object and assign them to their own variables
 
 
 
 2. What is a DOM event?
 
-  Your answer:
+  Your answer: A DOM event occurs when code allows a user input to update the DOM
 
-  Researched answer:
+  Researched answer: // Had some trouble finding an answer that made sense to me //
 
 
 
 3. What is the difference between a class and an object?
 
-  Your answer:
+  Your answer: A class is a blueprint for which objects are created on
 
-  Researched answer:
+  Researched answer: Looking into it a little more I discovered that a class is a template for objects. A class defines object properties including a valid range of values, and a default value. A class also describes object behavior. An object is a member or an "instance" of a class. An object has a state in which all of its properties have values that you either explicitly define or that are defined by default settings.
 
 
 
 4. What is the difference between a HTML div and a span?
 
-  Your answer:
+  Your answer: I haven't leanred about a span in HTML but a div is a basic container where text can be added and targeted. 
```

Comment:
Almost. A div is a block container element that holds nest/child elements within it. Think of DIV as a tupperware container that can hold more containers or items inside of it. 

A span is a inline element, mostly used with text elements (such as paragraphs, head1-6, etc) that allows you to target specific sections of the text and do something with the targeted area without having to break the element up into different tags.